### PR TITLE
fix: fix setBean single binding write and add tests

### PR DESCRIPTION
**Pull request by tepi**
Fixes https://github.com/vaadin/flow/issues/18869

Adds missing test to confirm that setBean works as before.

### CODE REVIEW

File: flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java [hunk 1]
```diff
@@ -2229,9 +2217,13 @@ public <FIELDVALUE> Binding<BEAN, FIELDVALUE> bindReadOnly(
...
-     * Any change made in one of the fields also runs validation for all the
-     * fields {@link Binding} and bean level validation for this binder (bean
-     * level validators are added using {@link Binder#withValidator(Validator)}.
+     * Note: Any change made in one of the bound fields runs validation for only
+     * the changed {@link Binding}, and additionally any bean level validation
+     * for this binder (bean level validators are added using
+     * {@link Binder#withValidator(Validator)}. As a result, the bean set via
+     * this method is not guaranteed to always be in a valid state. If bean
+     * validity is required at all times, {@link #readBean(Object)} and
+     * {@link #writeBean(Object)} should be used instead.
```
Comment (mcollovati) [inline flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java:2230]: Should we also add a note about `StatusChangeListener` and `BinderValidationStatusHandler` being invoked with validation successful events, even if the bean may be in an invalid stated?

File: flow-data/src/test/java/com/vaadin/flow/data/binder/BinderTest.java
Comment (mcollovati) [inline flow-data/src/test/java/com/vaadin/flow/data/binder/BinderTest.java:1]: Related to the comment about validation events, should we add assertions (or separated tests) to verify that valid status events are fired with `setBean` even if other fields are invalid?

File: flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java (hunk 1 above)
Comment (tepi) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java:2230]: Good point, I'll add it

File: flow-data/src/test/java/com/vaadin/flow/data/binder/BinderTest.java
Comment (tepi) [reply flow-data/src/test/java/com/vaadin/flow/data/binder/BinderTest.java:1]: That makes sense, I'll add those too

File: flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java (hunk 1 above)
Comment (tepi) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java]: Done

File: flow-data/src/test/java/com/vaadin/flow/data/binder/BinderTest.java
Comment (tepi) [reply flow-data/src/test/java/com/vaadin/flow/data/binder/BinderTest.java:1]: Done

**Review by mcollovati — APPROVED** [review]

**Pull request merged**